Src/FluentAssertions/Common/TypeExtensions.cs

```diff
@@ -194,46 +197,62 @@ public static FieldInfo FindField(this Type type, string fieldName, Type preferr
                 : properties.SingleOrDefault();
         }
 
-        public static IEnumerable<MemberInfo> GetNonPrivateMembers(this Type typeToReflect)
+        public static IEnumerable<MemberInfo> GetNonPrivateMembers(this Type typeToReflect, MemberVisibility visibility)
         {
             return
-                GetNonPrivateProperties(typeToReflect)
-                    .Concat<MemberInfo>(GetNonPrivateFields(typeToReflect))
+                GetNonPrivateProperties(typeToReflect, visibility)
+                    .Concat<MemberInfo>(GetNonPrivateFields(typeToReflect, visibility))
                     .ToArray();
         }
 
-        public static IEnumerable<PropertyInfo> GetNonPrivateProperties(this Type typeToReflect,
-            IEnumerable<string> filter = null)
+        public static IEnumerable<PropertyInfo> GetNonPrivateProperties(this Type typeToReflect, MemberVisibility visibility)
         {
             IEnumerable<PropertyInfo> query =
-                from propertyInfo in GetPropertiesFromHierarchy(typeToReflect)
+                from propertyInfo in GetPropertiesFromHierarchy(typeToReflect, visibility)
                 where HasNonPrivateGetter(propertyInfo)
                 where !propertyInfo.IsIndexer()
-                where filter is null || filter.Contains(propertyInfo.Name)
```

Comment:
Was a `filter` never provided by any caller?
Is the functionality handled elsewhere since it was originally added?

Reply:
I tried to look back until 4.x and it wasn't even used then.